language-design.md

```diff
@@ -28,7 +32,9 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+Steele provides great insight on the qualities of a well-designed language: “I should not design a small language, and I should not design a large one. I need to design a language that can grow. I need to plan ways in which it might grow -- but I need, too, to leave some choices so that other persons can make those choices at a later time” [Steele, 1998]. Steele stresses two major points which we felt were fundamental for a well-designed language: scalability and user involvement. A well-designed language should start small with a plan for growth. As its users become accustomed to it, their wants and needs may expand and the language must have a plan to grow to meet these needs. The language designers must find a balance between the Cathedral approach, in which the developer decides on all design choices, and the bazaar approach, in which the developer relies on user feedback for all design choices [Steele, 1998]. A well-designed language is not one that necessarily is perfect from conception, but rather one that provides a foundation for continual improvement. Steele makes a compelling argument for user involvement by invoking a well known fishing metaphor with a twist: while it is useful to teach a man to fish rather than give him a fish, it would be even better to provide him with the tools necessary to create fishing poles such that he can help others catch fish [Steele, 1998]. In the same sense, a well designed language should provide enough working vocabulary for users to seamlessly expand the language. For instance, when adding additional features to a language, the syntax for doing so should be similar enough such that the user does not need to more or less learn a new language. 
```

Comment:
I thoroughly agree that scalability and user involvement are very important qualities for a well-designed language to have. I'd like to add that I think scalability is important not just because the wants and needs of users may expand as they use it, _per se_. Aside from cases where users realize it would be much easier to do most tasks with features that are missing, there are cases where computers can suddenly do things they couldn't before (connecting over the internet and multi-threading come to mind) and users now want and need to program computers to do those things. I didn't mention this in my parts of my team's responses, but Steele makes a point to this effect, that "users will not now with glad cries glom on to a language that gives them no more than what Scheme or Pascal gave them."